hail/src/main/scala/is/hail/backend/local/LocalBackend.scala

```diff
@@ -68,7 +78,10 @@ class LocalBackend(
   def parallelizeAndComputeWithIndex(backendContext: BackendContext, collection: Array[Array[Byte]],
     dependency: Option[TableStageDependency] = None)(f: (Array[Byte], Int) => Array[Byte]): Array[Array[Byte]] = {
     collection.zipWithIndex.map { case (c, i) =>
-      f(c, i)
+      HailTaskContext.setTaskContext(new LocalTaskContext())
+      val bytes = f(c, i)
+      HailTaskContext.unset()
+      bytes
```

Comment:
@tpoterba I think this PR is almost done. I'm not sure if this is the right way to handle the `HailTaskContext` setting and unsetting on the `LocalBackend`. Does `LocalBackend` even need a TaskContext analogue to hold a `RegionPool`? Or should we just be doing like `Backend.get().getRegionPool()` and the local one just holds onto one whereas the Spark one checks the task context?

Reply:
> Does LocalBackend even need a TaskContext analogue to hold a RegionPool?

I think using similar patterns in all the backends will make it much easier for us to think about what the code is doing. I can certainly imagine that we'll want to use the local backend to interrogate per-task memory usage, for instance, with which the task context will help.